sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -49,7 +50,7 @@ import org.apache.spark.tags.ExtendedSQLTest
  *
  * To run a single test file upon change:
  * {{{
- *   build/sbt "~sql/test-only *SQLQueryTestSuite -- -z inline-table.sql"
```

Comment:
Not sure about this because practically it was good since file names can be matched. Now it cannot be for cosmetic reasons.

I added a clue in the PR I pointed out in order to make debugging easier with the Jenkins output in Github. So currently people at least can identify which file was failed.

Reply:
Huge win of this workaround is availability of looking into history of individual test. It would provide context which cannot be retrieve from test log, which is really helpful to track down test flakiness.
e.g. https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111332/testReport/org.apache.spark.sql/SQLQueryTestSuite/typeCoercion_native_mapconcat/history/

Full tests in SQLQueryTestSuite:
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111332/testReport/org.apache.spark.sql/SQLQueryTestSuite/

Full tests in ThriftServerQueryTestSuite (same test can be matched with SQLQueryTestSuite):
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111332/testReport/org.apache.spark.sql.hive.thriftserver/ThriftServerQueryTestSuite/

So I'm afraid I'm not sure it's just only cosmetic issue. This workaround brings actual benefits, though it has pros and cons, unfortunately. I guess someone would be familiar with file path instead of file path excluding ".sql", so please treat this as a proposal and weigh on current state vs "workaround applied" state.

Reply:
But this affects existing test way. Can't we find a way to override SBT's XML reporter or to use ScalaTest's XML reporter?

Fixing this in SBT, and upgrading the version should be the standard approach.

Reply:
I totally agree that's the way to go, but if no one would like to go forward, that's just an ideal approach. (And that doesn't seem easy to fix the sbt issue with ensuring it doesn't break anything.) sbt/sbt#2949 is filed at Feb. 2017, which is over 2 years ago.

Reply:
Yes. For the current approach in this PR, I am still not sure due to the downside of changing the existing test way.

I don't think many people actually know `build/sbt "~sql/test-only *SQLQueryTestSuite -- -z xxx.sql"` is based upon its test name and matching it to the exact file name is pretty good one.

Reply:
> I don't think many people actually know build/sbt "~sql/test-only *SQLQueryTestSuite -- -z xxx.sql" is based upon its test name

I'm not sure I could agree with this, as that means this feature is blindly used at all which it shouldn't (as it only applies to `*SQLQueryTestSuite`). This is even documented in Spark website. Please refer "Testing with SBT" in http://spark.apache.org/developer-tools.html

> matching it to the exact file name is pretty good one.

Yes I totally agree it is more intuitive and better. So that's "trade-off". Test history of *SQLQueryTestSuite in Jenkins is just unusable as of now, so personally I think it's worth to weigh on both. Not a strong opinion as I commented before - it's just a proposal.